src/sqlparser/src/keywords.rs

```diff
@@ -599,6 +599,7 @@ pub const RESERVED_FOR_TABLE_ALIAS: &[Keyword] = &[
     Keyword::SET,
     Keyword::RETURNING,
     Keyword::EMIT,
+    Keyword::FORMAT,
```

Comment:
https://www.postgresql.org/docs/current/sql-keywords-appendix.html
> FORMAT | non-reserved
> WITH | reserved, requires AS

This suggests our syntax extension introduces a conflict (#10268) / incompatibility with PostgreSQL. The following is valid in PostgreSQL but no longer in RisingWave:
```
select * from t format;
```

Reply:
Yes, but we have published the `FORMAT ... ENCODE ...` for a time and breaking that(such as move the `FORMAT` clause foward to the sink name) sounds unacceptable to me. Maybe we need some place to not those inconsistent behaviors with PG?
https://risingwave-labs.slack.com/archives/C034XCYJPSN/p1703218033241229

Reply:
Right, moving `format` closer to sink name is worse, as it is incompatible / inconsistent with our own. It was not a problem until sink into table, because prior `create sink` requires a `with (connector = )` clause before `format`. That `with` (reserved) helped avoid syntax conflict, but can be absent here 😢 As a result:
```
create sink s0 as select * from t with (...) format ...; -- no problem, `with` cannot be alias of `t`
create sink s1 into dest as select * from t format ...; -- pg would treat `format` as alias of `t`
```